packages/tangle/approvalweightmanager.go

```diff
@@ -433,39 +383,81 @@ func (a *ApprovalWeightManager) updateBranchWeight(branchID ledgerstate.BranchID
 	})
 }
 
-func (a *ApprovalWeightManager) moveMessageWeightToNewBranch(messageID MessageID, _, newBranchID ledgerstate.BranchID) {
+// processForkedMessage updates the Branch weight after an individually mapped Message was forked into a new Branch.
+func (a *ApprovalWeightManager) processForkedMessage(messageID MessageID, forkedBranchID ledgerstate.BranchID) {
 	a.tangle.Storage.Message(messageID).Consume(func(message *Message) {
-		a.propagateSupportToBranches(newBranchID, message)
+		a.tangle.Storage.BranchSupporters(forkedBranchID, NewBranchSupporters).Consume(func(forkedBranchSupporters *BranchSupporters) {
+			a.tangle.LedgerState.Branch(forkedBranchID).Consume(func(forkedBranch ledgerstate.Branch) {
+				if !a.addSupportToForkedBranchSupporters(identity.NewID(message.IssuerPublicKey()), forkedBranchSupporters, forkedBranch.(*ledgerstate.ConflictBranch).Parents(), message.SequenceNumber()) {
+					return
+				}
+
+				a.updateBranchWeight(forkedBranchID)
+			})
+		})
 	})
 }
 
 // take everything in future cone because it was not conflicting before and move to new branch.
-func (a *ApprovalWeightManager) moveMarkerWeightToNewBranch(marker *markers.Marker, oldBranchID, newBranchID ledgerstate.BranchID) {
-	a.migrateMarkerSupportersToNewBranch(marker, oldBranchID, newBranchID)
+func (a *ApprovalWeightManager) processForkedMarker(marker *markers.Marker, oldBranchIDs ledgerstate.BranchIDs, forkedBranchID ledgerstate.BranchID) {
+	branchVotesUpdated := false
+	a.tangle.Storage.BranchSupporters(forkedBranchID, NewBranchSupporters).Consume(func(branchSupporters *BranchSupporters) {
+		a.tangle.LedgerState.Branch(forkedBranchID).Consume(func(forkedBranch ledgerstate.Branch) {
+			parentBranchIDs := forkedBranch.(*ledgerstate.ConflictBranch).Parents()
```

Comment:
Here should go an extensive comment on why the parents are the important thing here